app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class LikesController < ApplicationController
+  def create
+    post_id = params[:post_id]
+    user = Post.find(post_id).author
+    like = Like.new(author_id: current_user.id, post_id:)
+    if like.save
+      flash[:success] = 'Liked'
+    else
+      flash.now[:error] = "Error: Can't like this post"
+    end
+    redirect_to user_posts_path(user.id)
+  end
```

Comment:
- [ ] Also when  itry to like a post, I get this error in the screenshot below, referencing an unknown attribute `author_id`. Just as I suggested above, take a look at your model and very their relationship and `foreign_key` :+1: :airplane: 

- - Screenshot reference:
![Screenshot from 2023-11-02 15-57-53](https://github.com/JuanDiegoGuerra/Blog-App-RoR/assets/107632780/99bcc1a6-273b-4c04-b5e4-7f8d43e828bb)